util/util-internal-validation/src/dsl.ts

```diff
@@ -17,15 +17,8 @@ import {GetCastType, GetSrcType, Validator} from './interface'
 
 export function object<Props extends Record<string, Validator<any> | undefined>>(
     props: Props
-): Validator<GetPropsCast<Props>, GetPropsSrc<Props>> {
-    let presentProps: Record<string, Validator<any>> = {}
-    for (let key in props) {
-        let v = props[key]
-        if (v) {
-            presentProps[key] = v
-        }
-    }
-    return new ObjectValidator(presentProps) as any
+): ObjectValidator<Props> {
```

Comment:
idk how to tell to ts compiler that this function has to return the same type that is passed to the function but without undefined values 🤷‍♂️